fix: parse $refs containing the hash symbol correctly (#1445)
Redocly/redocly-cli: .changeset/slimy-planes-deliver.md

---
"@redocly/openapi-core": patch
"@redocly/cli": patch
---

> Fixed resolving $refs. Previously, it was failing to correctly resolve file names that contain the hash symbol.

Reviewer: ```suggestion
Fixed a bug with resolving $refs to file names that contain the hash symbol.
```
